### PR TITLE
Factory design pattern

### PR DESCRIPTION
**Pull request by Priyanka-droid**
[design pattern](https://sourcemaking.com/design_patterns)

### CODE REVIEW

**Comment by crazyscoop** [comment]
Add a reviewer and PR description.

Rename pr to `factory design pattern`.

Mention references in the description.

File: FactoryDesignPattern.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import java.util.Scanner;
+
+public class FactoryDesignPattern {
+    public static void main(String[] args) {
+        Logistics l= new Logistics();
```
Comment (crazyscoop) [inline FactoryDesignPattern.java]: use a code formatter plugin to format and beautify code.
for [vscode](https://marketplace.visualstudio.com/items?itemName=redhat.java)

File: FactoryDesignPattern.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                throw new Exception("go deliver it on your back!!!!");
+            }
+        }
+        System.out.println(t.findDeliveryPrice());
+        t.showDeliveryMessage();
+
+    }
+}
+
+//creator class
```
Comment (crazyscoop) [inline FactoryDesignPattern.java]: create a readme, add a simple description for the following
- creator
- product
- concreate creator etc.

**Comment by crazyscoop** [comment]
Keep separate folders for different patterns, add a markdown readme in the folder with important points.

File: FactoryDesignPattern.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+//concrete CreatorA
+class TruckLogistic extends TransportLogistic{
+     Mode createLogistic(){
+        return new Truck();
+    }
+}
+//concrete creatorB
+class ShipLogisticc extends TransportLogistic{
```
Comment (crazyscoop) [inline FactoryDesignPattern.java]:
```suggestion
class ShipLogistic extends TransportLogistic{
```

File: FactoryDesignPattern.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+
+    }
+}
+
+class Logistics {
+    void dealWithLogistic() throws Exception{
+        Scanner sc= new Scanner(System.in);
```
Comment (crazyscoop) [inline FactoryDesignPattern.java]: Scanner object needs to be closed after use.

File: FactoryDesignPattern.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        String transportMode= sc.nextLine();
+
+        TransportLogistic t;
```
Comment (crazyscoop) [inline FactoryDesignPattern.java]: Why do you need this abstract class, what if it was `Mode m` and you returned the concrete product classes i.e Ship and Truck below in the switch block?

**Review by crazyscoop — CHANGES_REQUESTED** [review]
check comments.

**Comment by crazyscoop** [comment]
[design pattern](https://sourcemaking.com/design_patterns)

File: FactoryDesignPattern.java (hunk 1 above)
Comment (Priyanka-droid) [reply FactoryDesignPattern.java]: done

File: FactoryDesignPattern.java (hunk 2 above)
Comment (Priyanka-droid) [reply FactoryDesignPattern.java]: done

File: FactoryDesignPattern.java (hunk 3 above)
Comment (Priyanka-droid) [reply FactoryDesignPattern.java]: Done

File: FactoryDesignPattern.java (hunk 4 above)
Comment (Priyanka-droid) [reply FactoryDesignPattern.java]: Done

**Review by crazyscoop — APPROVED** [review]
LGTM

**Pull request merged**